src/main/java/org/jenkinsci/test/acceptance/docker/DynamicDockerContainer.java

```diff
@@ -1,36 +0,0 @@
-package org.jenkinsci.test.acceptance.docker;
-
-import java.io.File;
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-
-import org.apache.commons.io.FileUtils;
-
-/**
- * A DockerContainer whose <code>Dockerfile</code> is not static but requires some knowledge of the environment it is going to run in.
- *
- * Before the image is created {@link #process(String)} will be called which can be used to replace place holders with given values or manipulate the docker file in anyother way.
- */
-public class DynamicDockerContainer extends DockerContainer {
```

Comment:
From what I can tell, @jtnord introduced this without any actual use cases, in #84. I have no opinion about it, and whether it should stay or be deleted seems orthogonal to this refactoring.